internal/smartyresponseprocessing/smartyresponseprocessing.go

```diff
@@ -0,0 +1,81 @@
+package smartyresponseprocessing
+
+import (
+	"encoding/json"
+	"fmt"
+)
+
+type SmartyResponse struct {
+	Zipcodes []struct {
+		Zipcode           string  `json:"zipcode"`
+		ZipcodeType       string  `json:"zipcode_type"`
+		DefaultCity       string  `json:"default_city"`
+		CountyFIPS        string  `json:"county_fips"`
+		CountyName        string  `json:"county_name"`
+		StateAbbreviation string  `json:"state_abbreviation"`
+		State             string  `json:"state"`
+		Latitude          float64 `json:"latitude"`
+		Longitude         float64 `json:"longitude"`
+		Precision         string  `json:"precision"`
+		AlternateCounties []struct {
+			CountyFIPS        string `json:"county_fips"`
+			CountyName        string `json:"county_name"`
+			StateAbbreviation string `json:"state_abbreviation"`
+			State             string `json:"state"`
+		} `json:"alternate_counties"`
+	} `json:"zipcodes"`
+}
+
+type ZipcodeResult struct {
+	Zipcode         string
+	USPSFips        []string
+	SmartyFips      []string
+	Inconsistencies int
+}
+
+func ProcessSmartyResponse(responseBody []byte, zipToCounties map[string][]string, yield func(ZipcodeResult)) error {
+	var smartyResponses []SmartyResponse
+	err := json.Unmarshal(responseBody, &smartyResponses)
+	if err != nil {
+		return fmt.Errorf("failed to unmarshal response: %w", err)
+	}
+
+	for _, response := range smartyResponses {
```

Comment:
If Smarty thinks a ZIP is invalid, it returns a response of this form ([docs](https://www.smarty.com/docs/cloud/us-zipcode-api)):

```
{
  "input_index": 0,
  "status": "invalid_zipcode",
  "reason": "Invalid ZIP Code."
}
```

We probably want to catch this response and either report it (missing ZIP) or consider it an empty list of counties for the given ZIP.